fix(kafka): fix template processing for header (#11527)
emqx/emqx: changes/ee/fix-11527.en.md

> Fixed an issue on Kafka header handling when placeholder translates to an array of key-value pairs on format `[{"key": "foo", "value": "bar"}]`.

Reviewer: ```suggestion
Fixed an issue with Kafka header handling when placeholders resolve to an array of key-value pairs (e.g.: `[{"key": "foo", "value": "bar"}]`).
```
